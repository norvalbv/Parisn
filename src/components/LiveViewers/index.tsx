import clsx from 'clsx';
import React, { ReactElement, useEffect, useState } from 'react';
import { io } from 'socket.io-client';
import { FontSize, fontSizeMap } from 'types';

const socket = io('ws://localhost:8000', {
  withCredentials: true,
});

export type LiveViewersProps = {
  className?: string;
  fontSize?: FontSize;
  label?: string | JSX.Element;
  /**
   * Params for page
   */
  pageParams?: string;
  /**
   * Other params, i.e, product id.
   */
  params?: string;
};

const LiveViewers = ({
  className,
  fontSize = 'xs',
  label = 'Live viewers:',
  pageParams,
  params,
}: LiveViewersProps): ReactElement => {
  const [viewCount, setViewCount] = useState(0);

  useEffect(() => {
    if (!params) {
      socket.emit('get room count', pageParams, (amount: number) => {
        return setViewCount(amount);
      });
    } else {
      socket.emit('get all room counts', params, (amount: number) => {
        return setViewCount(amount);
      });
    }
  }, [pageParams, params]);

  return (
    <p className={clsx(fontSizeMap[fontSize], className)}>
      {label} {viewCount}
    </p>
  );
};

export default React.memo(LiveViewers);
