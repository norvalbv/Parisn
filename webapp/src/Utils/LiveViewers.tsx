import React, { useState } from 'react';
import io from 'socket.io-client';

let socket = io('ws://localhost:8000', {
  withCredentials: true,
});

const LiveViewers = () => {
  const [viewCount, setViewCount] = useState(0);

  socket.on('get viewers', (amount) => setViewCount(amount));

  return (
    <div>
      <p>Live viewers: {viewCount}</p>
    </div>
  );
};

export default LiveViewers;
