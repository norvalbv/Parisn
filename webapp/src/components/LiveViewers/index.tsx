import React, { useEffect, useState } from 'react';
import { useSearchParams } from 'react-router-dom';
import { io } from 'socket.io-client';

let socket = io('ws://localhost:8000', {
  withCredentials: true,
});

export interface LiveViewersProps {
  pageParams?: string;
  params?: string;
}

const LiveViewers = ({ params, pageParams }: LiveViewersProps) => {
  const [viewCount, setViewCount] = useState(0);

  useEffect(() => {
    if (!params) {
      socket.emit('get room count', pageParams, (amount: number) => {
        return setViewCount(amount);
      });
    } else {
      socket.emit('get all room counts', params, (amount: number) => {
        return setViewCount(amount);
      });
    }
  }, []);

  return <p className="text-sm z-10">Live viewers: {viewCount}</p>;
};

export default React.memo(LiveViewers);
