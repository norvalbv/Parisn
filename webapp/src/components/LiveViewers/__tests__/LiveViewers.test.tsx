import { FC } from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import LiveViewers, { LiveViewersProps } from '../';

const LiveViewersWithRequiredProps: FC<Required<LiveViewersProps>> = LiveViewers;

describe('<Radial />', () => {
  test('Radial props with mandatory props', async () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LiveViewers />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Radial props with all props as required', async () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LiveViewersWithRequiredProps
            params="test-params=test"
            fontSize="xl"
            label="Test Label"
            pageParams="/test"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
