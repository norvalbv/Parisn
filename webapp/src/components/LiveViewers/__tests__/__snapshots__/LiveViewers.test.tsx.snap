// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radial /> Radial props with all props as required 1`] = `
<p
  className="text-xl bg-red-500 z-10"
>
  Test Label
   
  0
</p>
`;

exports[`<Radial /> Radial props with mandatory props 1`] = `
<p
  className="text-xs  z-10"
>
  Live viewers:
   
  0
</p>
`;
